python/lsst/meas/algorithms/subtractBackground.py

```diff
@@ -327,29 +341,30 @@ def fitBackground(self, maskedImage, nx=0, ny=0, algorithm=None):
             if self.config.approxOrderY not in (self.config.approxOrderX, -1):
                 raise ValueError("Error: approxOrderY not in (approxOrderX, -1)")
             order = self.config.approxOrderX
-            minNumberGridPoints = self.config.approxOrderX + 1
-            if min(nx, ny) <= self.config.approxOrderX:
+            minNumberGridPoints = order + 1
+            if min(nx, ny) <= order:
                 self.log.warn("Too few points in grid to constrain fit: min(nx, ny) < approxOrder) "
-                              "[min(%d, %d) < %d]" % (nx, ny, self.config.approxOrderX))
+                              "[min(%d, %d) < %d]" % (nx, ny, order))
                 if self.config.undersampleStyle == "THROW_EXCEPTION":
-                    raise ValueError("Too few points in grid (%d, %d) for order (%d) and binsize (%d)" % (
-                        nx, ny, self.config.approxOrderX, self.config.binSize))
+                    raise ValueError("Too few points in grid (%d, %d) for order (%d) and binSize (%d, %d)" %
+                                     (nx, ny, order, binSizeX, binSizeY))
                 elif self.config.undersampleStyle == "REDUCE_INTERP_ORDER":
                     if order < 1:
                         raise ValueError("Cannot reduce approxOrder below 0.  " +
                                          "Try using undersampleStyle = \"INCREASE_NXNYSAMPLE\" instead?")
                     order = min(nx, ny) - 1
                     self.log.warn("Reducing approxOrder to %d" % order)
                 elif self.config.undersampleStyle == "INCREASE_NXNYSAMPLE":
+                    # Reduce bin size to the largest acceptable square bins
```

Comment:
Do they have to be square?  The `order` is (unfortunately) identical for both dimensions, but I don't think there's a reason the bin size has to be identical.